core/src/modules/ops/record.ts

```diff
@@ -969,11 +984,46 @@ export namespace RecordOps {
       }
     );
 
+    const invalidRecords = await api.sequelize.query(
+      `
+      SELECT DISTINCT
+        "GrouparooRecord"."id"
+      FROM
+        "records" AS "GrouparooRecord"
+        INNER JOIN "recordProperties" AS "recordProperties" ON "GrouparooRecord"."id" = "recordProperties"."recordId"
+      WHERE
+        "GrouparooRecord"."invalid" = FALSE
+        AND "recordProperties"."invalidReason" IS NOT NULL
+    `,
```

Comment:
Please 👀

Reply:
I don't think you need to return anything here - this whole method should only be operating on the collection of `records` found by the first query.  That's the group of records found that are ready to go ready, limited by `limit`.

So this query's job is to update `record.invalid` for only those already found record IDs.

```ts
// psuedocode

await GrouparooRecord.update(
  {invalid:true},
  {where:{id:{in:  `select distinct recordId from recordProperties where recordId in PRV_RECORD_IDS and invalidReason IS NOT NULL` }}}
})

// and then 

await GrouparooRecord.update(
  {invalid:false},
  {where:{id:{in:  `select distinct recordId from recordProperties where recordId in PRV_RECORD_IDS and invalidReason IS NULL` }}}
})
```